awx/main/tasks.py

```diff
@@ -245,13 +248,17 @@ def apply_cluster_membership_policies():
 
         # Finally, process instance policy percentages
         for g in sorted(actual_groups, key=lambda x: len(x.instances)):
+            exclude_type = 'execution' if g.obj.name == settings.DEFAULT_CONTROL_PLANE_QUEUE_NAME else 'control'
+            candidate_pool_ct = len([i for i in actual_instances if i.obj.node_type != exclude_type])
             policy_per_added = []
             for i in sorted(actual_instances, key=lambda x: len(x.groups)):
+                if i.obj.node_type == exclude_type:
+                    continue
                 if i.obj.id in g.instances:
                     # If the instance is already _in_ the group, it was
                     # applied earlier via a minimum policy or policy list
                     continue
-                if 100 * float(len(g.instances)) / len(actual_instances) >= g.obj.policy_instance_percentage:
+                if 100 * float(len(g.instances)) / candidate_pool_ct >= g.obj.policy_instance_percentage:
```

Comment:
super corner case:
* you deploy awx with control nodes only (intending to deploy execution nodes dynamically later perhaps)
* you create a new IG
* `candidate_pool_ct` will be zero => divide by zero error

Reply:
if there are no candidate instances, we shouldn't ever be getting here, because that means we're considering whether to add an instance. We could add a `continue` in that case